src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAISickTask.java

```diff
@@ -0,0 +1,498 @@
+    }
+
+    /**
+     * Go to the hut to try to get food there first.
+     *
+     * @return the next state to go to.
+     */
+    private DiseaseState goToHut(final ICitizenData data)
+    {
+        final IBuildingWorker buildingWorker = data.getWorkBuilding();
+        if (buildingWorker == null)
+        {
+            return SEARCH_HOSPITAL;
+        }
+
+        if (citizen.getCitizenSleepHandler().isAsleep() || citizen.isWorkerAtSiteWithMove(buildingWorker.getPosition(), MIN_DIST_TO_HUT))
+        {
+            return SEARCH_HOSPITAL;
+        }
+        return GO_TO_HUT;
+    }
+
+    /**
+     * Go to the previously found placeToPath to get cure.
+     *
+     * @return the next state to go to.
+     */
+    private DiseaseState goToHospital()
+    {
+        if (placeToPath == null)
+        {
+            return SEARCH_HOSPITAL;
+        }
+
+        if (citizen.getCitizenSleepHandler().isAsleep() || citizen.isWorkerAtSiteWithMove(placeToPath, MIN_DIST_TO_HOSPITAL))
+        {
+            return WAIT_FOR_CURE;
+        }
+        return SEARCH_HOSPITAL;
+    }
+
+    /**
+     * Search for a placeToPath within the colony of the citizen.
+     *
+     * @param citizenData the citizen.
+     * @return the next state to go to.
+     */
+    private DiseaseState searchHospital(final ICitizenData citizenData)
+    {
+        final IColony colony = citizenData.getColony();
+        placeToPath = colony.getBuildingManager().getBestHospital(citizen);
+
+        if (placeToPath == null)
+        {
+            final String id = citizen.getCitizenDiseaseHandler().getDisease();
+            if (id.isEmpty())
+            {
+                return IDLE;
+            }
+            final Disease disease = IColonyManager.getInstance().getCompatibilityManager().getDisease(id);
+            citizenData.triggerInteraction(new StandardInteractionResponseHandler(new TranslationTextComponent(NO_HOSPITAL, disease.getName(), disease.getCureString()), new TranslationTextComponent(NO_HOSPITAL),
+              ChatPriority.BLOCKING));
+            return IDLE;
+        }
+
+        // Reset AI when starting to go to the hospital.
+        if (citizen.getCitizenJobHandler().getColonyJob() != null)
+        {
+            citizen.getCitizenJobHandler().getColonyJob().resetAI();
+        }
+        return GO_TO_HOSPITAL;
+    }
+
+    /**
+     * Checks if the citizen has the cure in the inventory and makes a decision based on that.
+     *
+     * @param citizenData the citizen to check.
+     * @return the next state to go to.
+     */
+    private DiseaseState checkForCure(final ICitizenData citizenData)
+    {
+        final String id = citizen.getCitizenDiseaseHandler().getDisease();
+        if (id.isEmpty())
+        {
+            if (citizen.getHealth() > SEEK_DOCTOR_HEALTH)
+            {
+                reset();
+                currentState = IDLE;
+                return IDLE;
+            }
+            return GO_TO_HUT;
```

Comment:
isnt/shouldnt be he cured if the disease is empty and go back to idle/reset?

Reply:
If disease is empty the only way for him to get there is if the health was too low. So in this case we check if he recovered health (then reset) and else he will try his way to his hut and then to the hospital.